docs/rules.rst

```diff
@@ -2,24 +2,599 @@
+     - No
+   * - retry
+     - If the module fails execution, retry it once, boolean value true|false. Default false
+     - No
+   * - retries
+     - If the module fails execution, the number of times to retry it. Integer value, default 0
+     - No
+   * - delay
+     - The retry interval, an integer value
+     - No
+   * - verbosity
+     - Verbosity level when running the module, a value between 1-4
+     - No
+
+post_event
+**********
+.. list-table::  Post an event to a running rule set in the rules engine
+   :widths: 25 150 10
+   :header-rows: 1
+
+   * - Name
+     - Description
+     - Required
+   * - event
+     - The event dictionary to post
+     - Yes
+   * - ruleset
+     - The name of the rule set to post the event, default is the current rule set name
+     - No
+
+Example::
+
+      action:
+        post_event:
+          ruleset: Test rules4
+          event:
+            j: 4
+
+Example, using data saved with assignment
+::
+
+      name: multiple conditions
+      condition:
+        all:
+          - events.first << event.i == 0
+          - events.second << event.i == 1
+          - events.third << event.i == events.first.i + 2 
+      action:
+        post_event:
+          ruleset: Test rules4
+          event:
+            data: "{{events.third}}"
+
+
+| The events and facts prefixes have rule scope and cannot be accessed outside of
+| rules. Please note the use of Jinja substitution when accessing the event results.
+
+assert_fact
+***********
+.. list-table:: Post a fact to the running rule set in the rules engine
+   :widths: 25 150 10
+   :header-rows: 1
+
+   * - Name
+     - Description
+     - Required
+   * - fact
+     - The fact dictionary to post
+     - Yes
+   * - ruleset
+     - The name of the rule set to post the fact, default is the current rule set name
+     - No
+
+Example
+::
+
+    action:
+        assert_fact:
+          ruleset: Test rules4
+          fact:
+            j: 1
+
+Example, using data saved with assignment in multiple condition
+::
+
+      name: multiple conditions
+      condition:
+        all:
+          - events.first << event.i == 0
+          - events.second << event.i == 1
+          - events.third << event.i == events.first.i + 2 
+      action:
+        assert_fact:
+          ruleset: Test rules4
+          fact:
+            data: "{{events.first}}"
+
+Example, using data saved with single condition
+::
+
+      name: single condition
+      condition: event.i == 23
+      action:
+        assert_fact:
+          fact:
+            myfact: "{{event.i}}"
+
+| A rulebook can have multiple rule sets, the assert_fact/retract_fact/post_event allow you
+| to target different rule sets within the rulebook. You currently cannot assert an event to
+| multiple rule sets, it can be asserted to a single rule set. The default being the current
+| rule set. Please not the use of Jinja substitution in the above examples  when accessing 
```

Comment:
```suggestion
| rule set. Please not the use of Jinja substitution in the above examples when accessing 
```

Reply:
@ttuffin Please review, I have added the FAQ section which we can evolve